packages/core/src/components/modal/test/open-close/modal.e2e.ts

```diff
@@ -0,0 +1,55 @@
+import { test } from 'stencil-playwright';
+import { expect } from '@playwright/test';
+
+const componentTestPath = 'src/components/modal/test/open-close/index.html';
+
+test.describe('tds-modal-open', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto(componentTestPath);
+  });
+
+  test('1. Modal is opened by default', async ({ page }) => {
+    const tdsModal = page.locator('tds-modal');
+    const hasShowAttribute = await tdsModal.getAttribute('show');
+    if (hasShowAttribute) {
```

Comment:
I'm not sure we need to use conditionals here.

Additionally if tds-modal will always have this show attribute if you set it in the index.html tag so it does not give a fair representation of a good test, lets say for example a developer would still be able to make the element have the attribute show without showing the modal.

Instead I would remove the getAttribute show and look for the modal and see if it has all its visible elements